src/components/flat-table/flat-table.stories.js

```diff
@@ -96,6 +103,132 @@ export const basic = () => {
+    { name: 'client', isActive: false },
+    { name: 'total', isActive: false }
+  ];
+
+  const bodyDataItems = [
+    { client: 'Jason Atkinson', total: 1349 },
+    { client: 'Monty Parker', total: 849 },
+    { client: 'Blake Sutton', total: 3840 },
+    { client: 'Tyler Webb', total: 280 }
+  ];
+  const [headData, setHeadData] = useState(headDataItems);
+  const [bodyData, setBodyData] = useState(bodyDataItems);
+  const [sortType, setSortType] = useState('asc');
+  const sortByNumber = (dataToSort, sortByValue, type) => {
+    const sortedData = dataToSort.sort((a, b) => {
+      if (type === 'asc') {
+        return a[sortByValue] - b[sortByValue];
+      }
+
+      if (type === 'desc') {
+        return b[sortByValue] - a[sortByValue];
+      }
+
+      return 0;
+    });
+
+    return sortedData;
+  };
+
+  const sortByString = (dataToSort, sortByValue, type) => {
+    const sortedData = dataToSort.sort((a, b) => {
+      const nameA = a[sortByValue].toUpperCase();
+      const nameB = b[sortByValue].toUpperCase();
+
+      if (type === 'asc') {
+        if (nameA < nameB) {
+          return -1;
+        }
+
+        if (nameA > nameB) {
+          return 1;
+        }
+      }
+
+      if (type === 'desc') {
+        if (nameA > nameB) {
+          return -1;
+        }
+
+        if (nameA < nameB) {
+          return 1;
+        }
+      }
+
+      return 0;
+    });
+
+    return sortedData;
+  };
+
+  const handleClick = (e) => {
+    const tempHeadData = headData;
+    tempHeadData.forEach((item) => {
+      item.isActive = false;
+      if (item.name === e) {
+        item.isActive = !item.isActive;
+      }
+    });
+
+    setHeadData([...tempHeadData]);
+
+    const sortByValue = e;
+    let sortedData;
+
+    if (typeof bodyData[0][sortByValue] === 'string') {
+      sortedData = sortByString(bodyData, sortByValue, sortType);
+    }
+
+    if (typeof bodyData[0][sortByValue] === 'number') {
+      sortedData = sortByNumber(bodyData, sortByValue, sortType);
+    }
+
+    setBodyData([...sortedData]);
+    setSortType(sortType === 'asc' ? 'desc' : 'asc');
+  };
+
+  const showSortIcon = (isActive) => {
+    return isActive && <Icon type={ sortType === 'asc' ? 'sort_down' : 'sort_up' } />;
+  };
+
+  return (
+    <FlatTable colorTheme={ colorTheme }>
+      <FlatTableHead>
+        <FlatTableRow>
+          {
+            headData.map((dataItem) => {
+              return (
+                <FlatTableSortHeader key={ dataItem.name } onClick={ () => handleClick(dataItem.name) }>
+                  {dataItem.name}
+                  {showSortIcon(dataItem.isActive)}
+                </FlatTableSortHeader>
+              );
+            })
+          }
+        </FlatTableRow>
+      </FlatTableHead>
+      <FlatTableBody>
+        {
+          bodyData.map((dataItem) => {
+            return (
+              <FlatTableRow key={ dataItem.client }>
+                <FlatTableRowHeader>{dataItem.client}</FlatTableRowHeader>
```

Comment:
Should this not be `<FlatTableCell>` as this is rendering more `<th>` elements inside the body of the table?

Reply:
fixed in 31b18094395e7e61623c393f23dcadb8a0a0a664